File: Day 24/day24.ts
type TreeNode2 = {
    value: number;
    left?: TreeNode2;
    right?: TreeNode2;
} | null;

function isTreesSynchronized(tree1: TreeNode2, tree2: TreeNode2): [boolean, string] {
    if (!tree1 && !tree2) {
        return [true, ""];
    }

    function searchEqualTree(t1: TreeNode2, t2: TreeNode2): boolean {
        if (!t1 && !t2) {
            return true;
        }
        
        if (!t1 || !t2 || t1.value !== t2.value) {
            return false;
        }

        return searchEqualTree(t1.left ?? null, t2.right ?? null) &&
               searchEqualTree(t1.right ?? null, t2.left ?? null);
    }
    
    return [searchEqualTree(tree1, tree2), tree1 ? String(tree1.value) : ""];
}

File: Day 24/day24.py
def is_trees_synchronized(tree1, tree2):
    if not tree1 and not tree2:
        return [True, ""]
    

    def search_equal_tree(tree1, tree2):
        if not tree1 and not tree2:
            return True
        
        if not 'left' in tree1:
            return tree1['value'] == tree2['value']
        else:
            return tree1['value'] == tree2['value'] and search_equal_tree(tree1['left'],tree2['right']) and search_equal_tree(tree1['right'],tree2['left'])
  

    return [search_equal_tree(tree1, tree2), tree1['value']]

